DailyCodingProblem/Program.cs

```diff
@@ -42,5 +44,51 @@ static void Problem1()
 
             Console.WriteLine($"Sorry! There is not two numbers in the list that add up to {number}");
         }
+
+
+        /// <summary>
+        /// Given an array of integers, return a new array such that each element of the new array is the product of all the numbers in the original array except the one at the current index.
+        /// </summary>
+        static void Problem2()
+        {
+            Console.WriteLine("I need you to give me at least five integers. Please enter one at the time!");
+            List<int> listOfNumbers = new List<int>();
+
+            while(listOfNumbers.Count < 5)
+            {
+                Console.Write("Please enter a number: ");
+                int number;
+                
+                while(!int.TryParse(Console.ReadLine(), out number))
+                {
+                    Console.Write("Your input is not valid. Please enter a integer: ");
+                }
+
+                listOfNumbers.Add(number);
+                
+            }
+
+            List<int> listOfProducts = new List<int>();
+
+            for (int i = 0; i < listOfNumbers.Count; i++)
+            {
+                int element = listOfNumbers[i];
+                listOfNumbers.RemoveAt(i);
+                int product = 1;
+                
+                for (int j = 0; j < listOfNumbers.Count; j++)
+                {
+                    product *= listOfNumbers[j];
+                }
+
+                listOfProducts.Add(product);
+                listOfNumbers.Insert(i, element);
```

Comment:
So what you are doing is totally working and correct, but it's very expensive on the computer to insert and remove elements from a sequential list. (Remind me to show you the difference between linked and sequential and why it is expensive for one and not the other)

So my suggestion would be to either:

- Keep your list as it is and just skip over the number you are on
- Change your list to a linked list